Travis support doesn't tell user to create API key for Heroku

##### **Overview of the issue**

When running `jhipster ci-cd`, and choosing Travis, you're prompted to deploy to Heroku, but no information is given that an API key is needed.

Here's the instructions for adding an API key: https://docs.travis-ci.com/user/deployment/heroku/

##### **Motivation for or Use Case**

Build fails in Travis with the following error:

```
[ERROR] Failed to execute goal com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy (default-cli)
 on project gallery: Failed to deploy application: Could not get API key! Please install the
 toolbelt and login with `heroku login` or set the HEROKU_API_KEY environment variable. 
```

##### **Reproduce the error**

Run `jhipster ci-cd`, choose Travis, select "Yes" for Heroku deployment (this might be dependent on integrating Heroku first), then turn on Travis for your repo and run the tests. 

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

Print or run the following command to let users know they need to add an encrypted API key to their `.travis.yml`:

```
travis encrypt $(heroku auth:token) --add deploy.api_key
```

##### **JHipster Version(s)**

```
gallery@0.0.0 /Users/mraible/dev/okta/okta-react-photo-gallery-example
└── (empty)

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.okta.developer",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.0.0",
    "applicationType": "monolith",
    "baseName": "gallery",
    "packageName": "com.okta.developer",
    "packageFolder": "com/okta/developer",
    "serverPort": "8080",
    "authenticationType": "oauth2",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "yarn",
    "testFrameworks": [
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "fr"
    ],
    "herokuDeployType": "git"
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Album (album) {
  title String required,
  description TextBlob,
  created Instant
}
entity Photo (photo) {
  title String required,
  description TextBlob,
  image ImageBlob required,
  height Integer,
  width Integer,
  taken Instant,
  uploaded Instant
}
entity Tag (tag) {
  name String required minlength(2)
}
relationship ManyToOne {
  Album{user(login)} to User,
  Photo{album(title)} to Album
}
relationship ManyToMany {
  Photo{tag(name)} to Tag{photo}
}

paginate Album with pagination
paginate Photo, Tag with infinite-scroll

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)

git version 2.14.1

node: v9.8.0

npm: 6.1.0

yeoman: 2.0.2

yarn: 1.7.0

Docker version 18.03.1-ce, build 9ee9f40

docker-compose version 1.21.1, build 5a3f1a3

--- a/generators/ci-cd/prompts.js
+++ b/generators/ci-cd/prompts.js
@@ -136,7 +136,7 @@ function askIntegrations() {
             when: (this.pipelines.includes('jenkins') || this.pipelines.includes('gitlab') || this.pipelines.includes('circle') || this.pipelines.includes('travis')) && this.herokuAppName,
             type: 'checkbox',
             name: 'heroku',
-            message: 'Deploy to heroku?',
+            message: 'Deploy to heroku (requires HEROKU_API_KEY set on CI service)?',
             default: [],
             choices: herokuChoices
         }

--- a/generators/ci-cd/templates/circle.yml.ejs
+++ b/generators/ci-cd/templates/circle.yml.ejs
@@ -67,7 +67,7 @@ test:
 <%_ } _%>
 <%_ if (buildTool === 'maven') { _%>
     <%_ if (heroku.includes('circle')) { _%>
-        - ./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
+        - ./mvnw com.heroku.sdk:heroku-maven-plugin:2.0.5:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
     <%_ } else { _%>
         - ./mvnw verify -Pprod -DskipTests
     <%_ } _%>

--- a/generators/ci-cd/templates/jenkins/Jenkinsfile.ejs
+++ b/generators/ci-cd/templates/jenkins/Jenkinsfile.ejs
@@ -88,7 +88,7 @@ node {
     <%_ } _%>
     <%_ if (heroku.includes('jenkins')) { _%>
 <%= indent %>    stage('package and deploy') {
-<%= indent %>        sh "./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>"
+<%= indent %>        sh "./mvnw com.heroku.sdk:heroku-maven-plugin:2.0.5:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>"
     <%_ } else { _%>
 <%= indent %>    stage('packaging') {
 <%= indent %>        sh "./mvnw verify -Pprod -DskipTests"

--- a/generators/ci-cd/templates/travis.yml.ejs
+++ b/generators/ci-cd/templates/travis.yml.ejs
@@ -83,7 +83,7 @@ script:
 <%_ } _%>
 <%_ if (buildTool === 'maven') { _%>
   <%_ if (heroku.includes('travis')) { _%>
-  - ./mvnw com.heroku.sdk:heroku-maven-plugin:1.1.1:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
+  - ./mvnw com.heroku.sdk:heroku-maven-plugin:2.0.5:deploy -DskipTests -Pprod -Dheroku.appName=<%= herokuAppName %>
   <%_ } else { _%>
   - ./mvnw verify -Pprod -DskipTests
   <%_ } _%>

